Azurekid - Update ASIM templates (ASIM)

azurekid opened the pull request:
Required items, please complete
   
   Change(s):
   - ASimAuthenticationTemplate.yaml
   - ASimDnsTemplate.yaml
   - ASimFileEventTemplate.yaml
   - ASimNetworkSessionTemplate.yaml
   - vimAuthenticationTemplate.yaml
   - vimDnsTemplate.yaml
   - NetworkSessionTemplate.yaml
   - vimWebSessionTemplate.yaml
   
   Added:
   - ASimAuditEventTemplate.yaml
   - ASimDhcpTemplate.yaml
   - ASimProcessEventTemplate.yaml
   - ASimRegistryEventTemplate.yaml
   - ASimUserManagementTemplate.yaml
   - vimAuditEventTemplate.yaml
   - vimProcessEventTemplate.yaml

   Reason for Change(s):
   - Fixed errors in files
   - Added missing parameters
   - Created new vim templates

   Version Updated:
   - N/A

   Testing Completed:
   - yes

v-sudkharat commented:
Hi @azurekid, thank you for raising Pull Request with us! We will review it and get back to you. Thanks!

azurekid commented:
Hi @vakohl and @v-sudkharat

I have updated some more files and added vim Parser templates for some products that don't have a vim schema defined yet.

To my opinion every ASIM log could be filtered on `starttime` and `endtime` as these are always required fields in the ASIM model.

To reflect the already existing parsers in the Microsoft Sentinel Repo and to have templates that are prepared for use in unified parsers, I have added the `disabled` parameter to all templates.

What is your take on these extra modifications?

v-sudkharat commented:
Hi @azurekid, thank you for raising Pull Request with us! We will review it and get back to you. Thanks!

vakohl commented:
Thanks @azurekid . I'll review them. FYI, we are working on applying filters for parser for schemas which doesn't have them like File, Registry etc.

azurekid commented:
> Thanks @azurekid . I'll review them. FYI, we are working on applying filters for parser for schemas which doesn't have them like File, Registry etc.

Sound cool! Any thing I can help with on contribute on?
Let me know, as I have some time available today 💪

vakohl commented:
> > Thanks @azurekid . I'll review them. FYI, we are working on applying filters for parser for schemas which doesn't have them like File, Registry etc.
> 
> Sound cool! Any thing I can help with on contribute on? Let me know, as I have some time available today 💪

@azurekid We will be adding filters for Registry, File and Authentication existing parsers. Feel free to raise PR in case you like to contribute. I can share the filter list with you.

azurekid commented:
> > > Thanks @azurekid . I'll review them. FYI, we are working on applying filters for parser for schemas which doesn't have them like File, Registry etc.
> > 
> > Sound cool! Any thing I can help with on contribute on? Let me know, as I have some time available today 💪
> 
> @azurekid We will be adding filters for Registry, File and Authentication existing parsers. Feel free to raise PR in case you like to contribute. I can share the filter list with you.

Yes please

vakohl commented:
> > > > Thanks @azurekid . I'll review them. FYI, we are working on applying filters for parser for schemas which doesn't have them like File, Registry etc.
> > > 
> > > 
> > > Sound cool! Any thing I can help with on contribute on? Let me know, as I have some time available today 💪
> > 
> > 
> > @azurekid We will be adding filters for Registry, File and Authentication existing parsers. Feel free to raise PR in case you like to contribute. I can share the filter list with you.
> 
> Yes please

@azurekid 
Authentication:

•	starttime
•	endtime
•	eventtype_in
•	eventresultdetails_in 
•	eventresult
•	TargetUsername_has_any 
•	TargetAppName_has_any
•	ActorUsername_has_any
•	srcipaddr_has_any_prefix
•	Srchostname_has_any
•	TargetIpAddr_has_any_prefix
•	dvcipaddr_has_any_prefix
•	DvcHostname_has_any
•	disabled

Registry:

•	starttime
•	endtime
•	eventtype_in
•	ActorUsername has_any
•	RegistryKey_has_any
•	RegistryValue_has_any
•	RegistryValueData_has_any
•	DvcHostname_has_any
•	disabled

File:

•	starttime
•	endtime
•	eventtype_in
•	srcipaddr_has_any_prefix
•	Actorusername_has_any
•	TargetFilePath_has_any
•	SrcFilePath_has_any
•	hashes_has_any
•	DvcHostname_has_any
•	disabled

azurekid commented:
> • starttime
> • endtime
> • eventtype_in
> • eventresultdetails_in
> • eventresult
> • TargetUsername_has_any
> • TargetAppName_has_any
> • ActorUsername_has_any
> • srcipaddr_has_any_prefix
> • Srchostname_has_any
> • TargetIpAddr_has_any_prefix
> • dvcipaddr_has_any_prefix
> • DvcHostname_has_any
> • disabled

@vakohl,

I see in other filtered parsers that some of the same parameter fields are lowercase instead of CamelCase

An example is:
FileEvent     - actorusername_has_any
Process Event - dvchostname_has_any

I assume the everything should be in lowercase to avoid any case sensitivity issues is the filtered parsers. Is that correct?

azurekid commented:
> • starttime
> • endtime
> • eventtype_in
> • srcipaddr_has_any_prefix
> • Actorusername_has_any
> • TargetFilePath_has_any
> • SrcFilePath_has_any
> • hashes_has_any
> • DvcHostname_has_any
> • disabled

updates are added to the templates

v-sudkharat commented:
Hi @azurekid, thank you for accommodating the suggested changes! We will review the Pull Request. Thanks!

azurekid commented:
Hey @v-sudkharat,

Anything I can do to keep this going?

vakohl commented on ASIM/dev/Parser YAML templates/vimProcessEventTemplate.yaml:
```diff
@@ -0,0 +1,94 @@
+    , targetusername_has:string        = '*'
+    , actorusername_has:string         = '*'
+    , dvcipaddr_has_any_prefix:dynamic = dynamic([])
+    , eventresult:string               = '*'
```
we dont have eventresult in process.
Please add:
hashes_has_any
eventtype
dvchostname_has_any

https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimProcessEvent/Parsers/vimProcessEventMicrosoft365D.yaml

azurekid replied:
Added the requested fields and validated these with the documentation.
`hashes_has_any` does not seem to be part of the ProcessEvent schema.
@vakohl please advice

vakohl replied:
@azurekid hashes_has_any filter can be checked against fields like TargetProcessMD5 , TargetProcessSHA1 etc.

vakohl commented on ASIM/dev/Parser YAML templates/vimRegistryEventTemplate.yaml:
```diff
@@ -0,0 +1,71 @@
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: eventtype_in
```
dynamic type

vakohl commented on ASIM/dev/Parser YAML templates/vimFileEventTemplate.yaml:
```diff
@@ -0,0 +1,76 @@
+Parser:
+  Title: File events ASIM parser for <product name>
```
filtering parser

vakohl commented on ASIM/dev/Parser YAML templates/vimProcessEventTemplate.yaml:
```diff
@@ -0,0 +1,94 @@
+Parser:
+  Title: Process event ASIM parser for <product name>
```
filtering parser

vakohl commented on ASIM/dev/Parser YAML templates/vimRegistryEventTemplate.yaml:
```diff
@@ -0,0 +1,71 @@
+Parser:
+  Title: Registry Event ASIM parser for <product name>
```
filtering parser

vakohl commented on ASIM/dev/Parser YAML templates/vimUserManagementTemplate.yaml:
```diff
@@ -0,0 +1,41 @@
+Parser:
+  Title: User Management activity ASIM parser for <product name>
```
filtering parser

vakohl commented on ASIM/dev/Parser YAML templates/vimUserManagementTemplate.yaml:
```diff
@@ -0,0 +1,41 @@
+    Default: datetime(null)
+  - Name: disabled
+    Type: bool
+    Default: false
```
missing filters. Please refer:
https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml

vakohl replied:
@azurekid I need to update ASIM documentation

v-sudkharat commented:
Hi @azurekid, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

azurekid commented:
> Hi @azurekid, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

Hi,
Had a very busy week last week.
Will pick this up on 2023/11/08

v-sudkharat commented:
Hi @azurekid, Ok, Thanks!

v-sudkharat commented:
Hi @azurekid, could you please have a look at above suggestions. Thanks!

azurekid commented on ASIM/dev/Parser YAML templates/ASimAuditEventTemplate.yaml:
```diff
@@ -0,0 +1,29 @@
+Parser:
+  Title: ASIM Audit Event parser for <product name>
+  Version: '<parser version>'
+  LastUpdated: <parser update date>
```
Added `<MMM DD, YYYY>` as done in other existing examples: November 13, 2023

azurekid commented on ASIM/dev/Parser YAML templates/vimAuditEventTemplate.yaml:
```diff
@@ -0,0 +1,76 @@
+    starttime:datetime                 = datetime(null)
+    , endtime:datetime                 = datetime(null)
+    , srcipaddr_has_any_prefix:dynamic = dynamic([])
+    , eventtype_in:string              = '*'
```
Good catch!

azurekid commented on ASIM/dev/Parser YAML templates/vimAuthenticationTemplate.yaml:
```diff
@@ -22,19 +22,58 @@ ParserParams:
+    Default: dynamic([])
+  - Name: dvchostname_has_any
+    Type: dynamic
+    Default: dynamic([])
```
Added filters for vimAuthenticationTemplate

azurekid commented on ASIM/dev/Parser YAML templates/vimNetworkSessionTemplate.yaml:
```diff
@@ -46,20 +43,34 @@ ParserParams:
   - Name: eventresult
     Type: string
     Default: "'*'"
```
removed double quotes

azurekid commented:
Hey @vakohl,

Thanks for the superb review of the code.
I have resolved most of the issues, but am awaiting your feedback on one comment.

vakohl commented:
> Hey @vakohl and @v-sudkharat
> 
> Thanks for the superb review of the code. I have resolved most of the issues, but am awaiting your feedback on one comment.

thank you @azurekid for approciating and for your invaluable contributions as well. I responded to your comments, please let me know if you still got more questions.
